### ISSUE TITLE
Undistort returning black images for one of the cameras for 3D project calibration

### ISSUE THREAD

**Issue by jwang945** [DeepLabCut/DeepLabCut#1128]
**Your Operating system and DeepLabCut version**
OS: Centos Linux 7
DeepLabCut Version: 2.1.8.2
Anaconda env used: yes

**Describe the problem**

Hello, I'm having some difficulties with the 3d calibration for two cameras. I input about 50 pairs of checkboard images and after iterating I get about 20 pairs that look good with correct corner detection. Here is a .zip of those images. However, after I run `deeplabcut.calibrate_cameras(config_path, cbrow=8, cbcol=6, calibrate=False, alpha=0.9)`, `deeplabcut.calibrate_cameras(config_path, cbrow=8, cbcol=6, calibrate=True, alpha=0.9)`, and then `deeplabcut.check_undistortion(config_path, cbrow=8, cbcol=6)`. Half of the images in the undistort folder are black, and the undistort looks incorrect. (.zip attached). 
[undistortion.zip](https://github.com/DeepLabCut/DeepLabCut/files/6051844/undistortion.zip)
[corners.zip](https://github.com/DeepLabCut/DeepLabCut/files/6051846/corners.zip)
[calibration_images.zip](https://github.com/DeepLabCut/DeepLabCut/files/6051848/calibration_images.zip)

If i continue onwards and run `deeplabcut.triangulate(config_path, '/home/jwang91/DeepLabCut/3dvideos', filterpredictions=False, videotype="mp4")`, the following error is returned.
<details><summary>Code output</summary><p>

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-9-2185cd04233c> in <module>
----> 1 deeplabcut.triangulate(config_path, '/home/jwang91/DeepLabCut/3dvideos', filterpredictions=False, videotype="mp4")

~/.conda/envs/deeplabcut/lib/python3.6/site-packages/deeplabcut/pose_estimation_3d/triangulation.py in triangulate(config, video_path, videotype, filterpredictions, filtertype, gputouse, destfolder, save_as_csv)
    212             #undistort points for this pair
    213             print("Undistorting...")
--> 214             dataFrame_camera1_undistort,dataFrame_camera2_undistort,stereomatrix,path_stereo_file = undistort_points(config,dataname,str(cam_names[0]+'-'+cam_names[1]),destfolder)
    215             if len(dataFrame_camera1_undistort) != len(dataFrame_camera2_undistort):
    216                 import warnings

~/.conda/envs/deeplabcut/lib/python3.6/site-packages/deeplabcut/pose_estimation_3d/triangulation.py in undistort_points(config, dataframe, camera_pair, destfolder)
    314     if True:
    315         # Create an empty dataFrame to store the undistorted 2d coordinates and likelihood
--> 316         dataframe_cam1 = pd.read_hdf(dataframe[0])
    317         dataframe_cam2 = pd.read_hdf(dataframe[1])
    318         scorer_cam1 = dataframe_cam1.columns.get_level_values(0)[0]

IndexError: list index out of range

</p></details>

**How to Reproduce the problem**
1.import dlc
2.create 3d project
3.move images into calibration_images folder
4.run calibrate_cameras(calibrate=False)
5.run calibrate_cameras(calibrate=True)
6.run check_undistort
7.see incorrect undistort folder

I would greatly appreciate any help or insight someone could provide for this! Thank you.

**Comment by MMathisLab** [comment]
please try with many different values of `alpha`, this changes the output dramatically, and needs to be set for every project. So, run step 5 above many times to see the results :)

**Comment by jwang945** [comment]
Thank you for your reply! I went ahead and tested out some varying alpha values for step 5. I went through 0 ... .1 ... .2 up to 1, and it seemed between 1 and .9 was most successful. I then went through .91 ... .92 ... to .99 and continued this way until i reached around .97, however, I am still not finding a alpha resulting in undistortion that looks correct. Do I need to continue and be more precise or could my initial images be the issue? Currently, the cameras are set up so they are orthogonal to each other.

**Comment by AlexEMG** [comment]
Can you maybe share some pictures?

**Comment by jwang945** [comment]
Hello, I haven't forgotten about this issue. i currently am having an issue with my VPN and can't access the folder with the undistort. I'm currently troubleshooting. Thank you for your patience.

**Comment by jwang945** [comment]
[undistortion_alpha.8.zip](https://github.com/DeepLabCut/DeepLabCut/files/6070694/undistortion_alpha.8.zip)
[undistortion_alpha.97.zip](https://github.com/DeepLabCut/DeepLabCut/files/6070695/undistortion_alpha.97.zip)
[undistortion_alpha.99.zip](https://github.com/DeepLabCut/DeepLabCut/files/6070696/undistortion_alpha.99.zip)
[undistortion_alpha.973.zip](https://github.com/DeepLabCut/DeepLabCut/files/6070697/undistortion_alpha.973.zip)

Above are a few example undistort folders at varying alpha values.
I did some additional testing with the MATLAB stereo calibration and I noticed that for all my images, it got the origins mixed up. Should I be using a checkerboard that has one odd side and one even side? images attached:
Mine:
<img width="1036" alt="Screen Shot 2021-02-28 at 10 44 39 PM" src="https://user-images.githubusercontent.com/47010726/109692057-108cd100-7b56-11eb-824a-7fbb585be22e.png">
Their Example:
<img width="760" alt="Screen Shot 2021-02-28 at 10 44 51 PM" src="https://user-images.githubusercontent.com/47010726/109692058-11256780-7b56-11eb-9c31-6659f9f20b03.png">

**Comment by MMathisLab** [comment]
>> Currently, the cameras are set up so they are orthogonal to each other.

Hi @jwang945 thanks for the the more info! So, as it's coded in DLC it requires they are in stereo view actually, not orthogonal

that is why the order is likely inverted in your plot above (i.e. their example is indeed stereo view)

But, I would try the triangulare step actually, curious to see what happens

![undistorted_points](https://user-images.githubusercontent.com/28102185/109702060-0d183a80-7b94-11eb-9b3c-3f9b81e15893.png)

**Comment by jwang945** [comment]
It seems it has no issue analyzing the videos, however it seems that the dataframe is empty when it tries to triangulate a 3d construction.
```IndexError                                Traceback (most recent call last)
<ipython-input-9-2185cd04233c> in <module>
----> 1 deeplabcut.triangulate(config_path, '/home/jwang91/DeepLabCut/3dvideos', filterpredictions=False, videotype="mp4")

~/.conda/envs/deeplabcut/lib/python3.6/site-packages/deeplabcut/pose_estimation_3d/triangulation.py in triangulate(config, video_path, videotype, filterpredictions, filtertype, gputouse, destfolder, save_as_csv)
    212             #undistort points for this pair
    213             print("Undistorting...")
--> 214             dataFrame_camera1_undistort,dataFrame_camera2_undistort,stereomatrix,path_stereo_file = undistort_points(config,dataname,str(cam_names[0]+'-'+cam_names[1]),destfolder)
    215             if len(dataFrame_camera1_undistort) != len(dataFrame_camera2_undistort):
    216                 import warnings

~/.conda/envs/deeplabcut/lib/python3.6/site-packages/deeplabcut/pose_estimation_3d/triangulation.py in undistort_points(config, dataframe, camera_pair, destfolder)
    314     if True:
    315         # Create an empty dataFrame to store the undistorted 2d coordinates and likelihood
--> 316         dataframe_cam1 = pd.read_hdf(dataframe[0])
    317         dataframe_cam2 = pd.read_hdf(dataframe[1])
    318         scorer_cam1 = dataframe_cam1.columns.get_level_values(0)[0]

IndexError: list index out of range

**Comment by MMathisLab** [comment]
are the camera names changed at some point? in the 3dconfig.yaml?

**Comment by jwang945** [comment]
my mistake! I had forgotten to rename my second video as with "camera-2" in the name. I swapped to some smaller videos and made sure they were named correctly and the triangulation call did not throw any errors. I was able to make a labeled 3d video, however it the labels aren't really there.
[1-21_DLC_3D.mpg.zip](https://github.com/DeepLabCut/DeepLabCut/files/6098167/1-21_DLC_3D.mpg.zip)

**Comment by jwang945** [comment]
Although, I actually just tested out making a 2D video, and it seems that the results are poor as well. I'm capture from a different camera and slightly different view and the model needs to be retrained. I'd be happy to post a follow up with a new mpg with a better scorer.

**Comment by jwang945** [comment]
Hello,

I recently retrained models for both camera views. Interestingly, it seems as though the triangulation was successful. I only did a preliminary labelling of about 20 frames each so the flicker is expected. I'll probably need to do far more labelling to get an improved scorer, however does it seem that the undistortion is correct? Is there any metric that I can use to verify?

Thank you.
[1-21_DLC_3D.mpg.zip](https://github.com/DeepLabCut/DeepLabCut/files/6151330/1-21_DLC_3D.mpg.zip)
